libimage/list_test.go

```diff
@@ -0,0 +1,54 @@
+package libimage
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestListWithEmptyLayerImage(t *testing.T) {
+	runtime, cleanup := testNewRuntime(t)
+	defer cleanup()
+	ctx := context.Background()
+	loadOptions := &LoadOptions{}
+	loadOptions.Writer = os.Stdout
+
+	list, err := runtime.CreateManifestList("m")
+	require.NoError(t, err)
+	require.NotNil(t, list)
+
+	for _, test := range []struct {
+		input       string
+		expectError bool
+		numImages   int
+		names       []string
+	}{
+		// OCI ARCHIVE
+		{"testdata/empty_image.tar", false, 1, []string{"localhost/i:latest"}},
+	} {
+		loadedImages, err := runtime.Load(ctx, test.input, loadOptions)
+		if test.expectError {
+			require.Error(t, err, test.input)
+			continue
+		}
+		require.NoError(t, err, test.input)
+		require.Equal(t, test.names, loadedImages, test.input)
+
+		// Make sure that all returned names exist as images in the
+		// local containers storage.
+		ids := []string{} // later used for image removal
+		names := [][]string{}
+		for _, name := range loadedImages {
+			image, resolvedName, err := runtime.LookupImage(name, nil)
+			require.NoError(t, err, test.input)
+			require.Equal(t, name, resolvedName, test.input)
+			ids = append(ids, image.ID())
+			names = append(names, image.Names())
+		}
```

Comment:
See the two lint failures of above.

Reply:
I had to remove the test because of this issue https://github.com/containers/common/pull/1651#issuecomment-1717159114, test was not able correctly test what I was expecting it to test.